goro/src/main/java/com/stanfy/enroscar/goro/BufferedGoroDelegate.java

```diff
@@ -0,0 +1,357 @@
+package com.stanfy.enroscar.goro;
+
+import java.util.ArrayList;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.stanfy.enroscar.goro.GoroFuture.IMMEDIATE;
+import static com.stanfy.enroscar.goro.Util.checkMainThread;
+
+/**
+ * Either delegates a call to a wrapped Goro or buffers invocation data.
+ */
+abstract class BufferedGoroDelegate extends Goro {
```

Comment:
Everything here is originally from `BoundGoroImpl` (except `updateDelegate` method).